pkg/pillar/types/zedroutertypes.go

```diff
@@ -1870,6 +1871,37 @@ type NetworkXObjectConfig struct {
 type IpRange struct {
 	Start net.IP
 	End   net.IP
+	Size  int
+}
+
+// Contains used to evaluate whether an IP address
+// is within the range
+func (ipRange IpRange) Contains(ipAddr net.IP) (bool, error) {
+	start := ipRange.Start.To4()
+	end := ipRange.End.To4()
+	cur := ipAddr.To4()
+	if cur == nil || start == nil || end == nil {
+		errStr := fmt.Sprintf("invalid configuration (%s, %s, %s)\n",
+			ipAddr.String(), ipRange.Start.String(), ipRange.End.String())
+		return false, errors.New(errStr)
+	}
+	if bytes.Compare(cur, start) >= 0 &&
+		bytes.Compare(cur, end) <= 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
+// HostCount used to count the number of hosts
+func (ipRange IpRange) HostCount() int {
+	addr1 := ipRange.Start.To4()
+	addr2 := ipRange.End.To4()
+	start := uint(addr1[0])<<24 + uint(addr1[1])<<16 +
+		uint(addr1[2])<<8 + uint(addr1[3])
+	end := uint(addr2[0])<<24 + uint(addr2[1])<<16 +
+		uint(addr2[2])<<8 + uint(addr2[3])
+	addrCount := int(end + 1 - start)
```

Comment:
Here you appear to assume that end is part of the range thus the range 128-255 will have room for 128 allocations, when in fact it only has room for 126 if I'm not mistaken.

Reply:
If you you subtract those to get the usable set somewhere else there should be some comments and perhaps other names. A range from 10-20 can fit 10 hosts AFAICT (or are the ranges always inclusive?), but a range from 0-255 can not fit 255 nor 256 since .255 is the broadcast address (and .0 is normally reserved). 
Thus the number of addresses might be end+1-start, but the number of hosts will be less in some cases.